tiledb/sm/array/array_directory.cc

```diff
@@ -1258,9 +1258,19 @@ Status ArrayDirectory::is_fragment(
     const std::unordered_set<std::string>& ok_uris_set,
     const std::unordered_set<std::string>& consolidated_uris_set,
     int* is_fragment) const {
+  // If parsing the fragment ID fails, it's not a fragment.
+  std::string name;
+  format_version_t fragment_format_version;
+  try {
+    FragmentID fragment_id{uri};
+    name = fragment_id.name();
+    fragment_format_version = fragment_id.array_format_version();
+  } catch (const std::exception& e) {
```

Comment:
I don't think it's a good idea to use try/catch as part of regular control flow, if you can add a function `is_fragment_uri` or something that would be better.

My main reason for this suggestion is interactive debugging. If you use `gdb` and do `catch throw` then it will break for each non-fragment file in the array.  This will be very disruptive without a way to automate inspecting the stack and continuing if it is caused here (which is not something I have ever figured out how to do)

Reply:
I did it this way in order to minimize changes to existing code and reduce risk if we backport this. Ideally we would have a factory function that returns an `optional`, but that would lose the error message. I thought of adding a separate `is_fragment_uri` function but decided against it because it would duplicate code and computations.